e2e/e2e_test.go

```diff
@@ -101,3 +109,12 @@ func TestBasicServiceNoOSM(t *testing.T) {
 		fixtures.NewClientDeployment(t, hostname, conf.TestNameservers),
 		svr, svc)
 }
+
+// TestPrometheus proves that users can consume Prometheus metrics emitted by our controllers
+func TestPrometheus(t *testing.T) {
+	t.Parallel()
+	tc := suite.StartTestCase(t)
+	ns := tc.NS()
+
+	tc.WithResources(append(fixtures.NewPrometheus(ns), fixtures.NewPrometheusClient(ns, conf.PromClientImage))...)
```

Comment:
does the prometheus instance not get deployed by default? shouldn't the add-on automatically take care of that?

Reply:
Which add-on would deploy prometheus? Nginx-ingresses emit metrics which can be consumed by prometheus. To prove that with an e2e test we need to deploy a prometheus instance so we can consume the metrics emitted by Nginx-ingresses.